packages/core/test/integer.test.ts

```diff
@@ -355,6 +356,121 @@ describe('Integer', () => {
       Integer.MAX_SAFE_VALUE.toString()
     )
   })
+
+  test('int(BigInt) should be reversed by Integer.toBigInt', () => {
+    fc.assert(
+      fc.property(
+        fc.bigInt({ max: Integer.MAX_SAFE_VALUE.toBigInt(), min: Integer.MIN_SAFE_VALUE.toBigInt() }),
+        bigint => bigint === int(bigint).toBigInt()
+      )
+    )
+  })
+
+  test('int(string) should be reversed by Integer.toString', () => {
+    fc.assert(fc.property(fc.integer().map(num => num.toString()), str => str === int(str).toString()))
+  })
```

Comment:
I think you could replace this single test with these two tests which in sum (due to transitivity of ===, which hopefully holds for every sane implementation of equality) asserts the same thing + more fine-grained transformations.

```suggestion
  test('int.toString() should match Integer.toString()', () => {
    fc.assert(fc.property(fc.integer(), i => i.toString() === int(i).toString()))
  })

  test('int(string) should match int(Integer)', () => {
    fc.assert(fc.property(fc.integer(), i => int(i) === int(i.toString())))
  })
```

This also covers the original test iff. we assume that `i === j` => `i.toString() === j.toString()` for `i`, `j` of type `int`. Alternatively, if we don't want to make that assumption, we can have all three tests or add a test along the lines of

```typescript
  test('int.toString() matches for equal numbers', () => {
    fc.assert(fc.property(fc.integer(), i => int(i).toString() === int(i).toString()))
  })
```
But yeah, that one is weird looking because it tests such a basic assumption.